File: 3100-water-bottles-ii/3100-water-bottles-ii.py
class Solution:
    def maxBottlesDrunk(self, numBottles: int, numExchange: int) -> int:
        drank = numBottles
        empty = numBottles

        while(numExchange > 0 and empty >= numExchange):
            drank += 1
            empty = (empty - numExchange + 1);
            numExchange += 1
        
        return drank

File: 3100-water-bottles-ii/3100-water-bottles-ii.cpp
class Solution {
public:
    int maxBottlesDrunk(int numBottles, int numExchange) {
        int drank = numBottles, empty = numBottles;
        
        while(numExchange > 0 && empty >= numExchange){
            drank++;
            empty = (empty - numExchange + 1);
            numExchange++;
        }
        return drank;
    }
};